test/gpu/native/studies/transpose/transpose.chpl

```diff
@@ -0,0 +1,98 @@
+use CTypes;
+use GPUDiagnostics;
+use GPU;
+use Time;
+
+config const numTrials = 10;
+config const useNaive = false;
+config const sizeX = 2048*8,
+             sizeY = 2048*8;
+config param blockSize = 16;
+
+pragma "codegen for GPU"
+pragma "always resolve function"
+export proc transposeMatrix(odata: c_ptr(real(32)), idata: c_ptr(real(32)), width: int, height: int) {
+  // Allocate extra columns for the shared 2D array to avoid bank conflicts.
+  var smVoidPtr = __primitive("gpu allocShared", numBytes(real(32))*(blockSize+1)*blockSize);
```

Comment:
Could you put `blockSize+1` in a `const paddedBlockSize` and use it instead. I imagine we might want to play with different paddings, it can make it easier to change the padding. Also would make the code more self documenting.

I use "padded", but if there is a more appropriate/specific term that you are aware of, feel free to use that.

Reply:
Oooh, and bonus points for making that offset a `config param`. :) The more knobs we can turn the better when it comes to performance tests.

Reply:
Can't make it a `const`, since the primitives require compile-time known constants (https://github.com/chapel-lang/chapel/issues/21422), but I can make it a param :)